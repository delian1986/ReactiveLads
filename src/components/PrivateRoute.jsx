import { Route, Redirect } from "react-router-dom";

function PrivateRoute({ component: Component, isLoggedIn, ...rest }) {
  return (
    <Route
      {...rest}
      render={(props) => {
        if (!isLoggedIn) {
          return (
            <Redirect to={{ pathname: "/login", state: { from: props.location } }} />
          );
        }

        return <Component {...props} />;
      }}
    />
  );
}

export { PrivateRoute };
