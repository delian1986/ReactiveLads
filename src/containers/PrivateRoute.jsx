import { getLoggedInStatus } from "../store/selectors";
import { PrivateRoute } from "../components/PrivateRoute";
import { connect } from "react-redux";

const mapStateToProps = (state) => {
  return {
    isLoggedIn: getLoggedInStatus(state)
  };
};

export default connect(mapStateToProps)(PrivateRoute);
