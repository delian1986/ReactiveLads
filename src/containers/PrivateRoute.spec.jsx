import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import PrivateRoute from "./PrivateRoute";

const mockStore = configureStore([]);

describe("tests <PrivateRoute/> component", () => {
  let store;
  let component;
  beforeEach(() => {
    store = mockStore({
      auth: {
        isLoggedIn: true
      }
    });
    store.dispatch = jest.fn();
    component = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <PrivateRoute component={() => <></>} />
        </Provider>
      </BrowserRouter>
    );
  });
  describe("Test rendering component", () => {
    it("should render with given state from Redux store", () => {
      expect(component.toJSON()).toMatchSnapshot();
    });
  });
});
