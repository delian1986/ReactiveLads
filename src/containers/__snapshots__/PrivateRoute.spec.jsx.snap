// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests <PrivateRoute/> component Test rendering component should render with given state from Redux store 1`] = `null`;
